Release 0.2 (#17)
engula/engula.github.io: docs/posts/release-0.2.md

# Release 0.2

> <h5>Dec 17, 2021</h5>

Reviewer: You can use a [yaml metadata](https://www.mkdocs.org/user-guide/writing-your-docs/#yaml-style-meta-data) for this purpose.

Reviewer: Sorry I get it wrong.

`git-revision-date` already adds a `Last update: December 15, 2021` footer, and

```
  - git-revision-date-localized:
      enable_creation_date: true
```

can add a footer:

```
Last update: December 15, 2021
Created: December 13, 2021
```

An elegant way to handle metadata should be considered later. Anyway, mkdocs is mainly aimed at writing docs so that is lightweight which also means less features.

Author: I also considered this solution. One problem is that it shows the date I created the post, which is "December 13, 2021" here, but that's not what we want. I am not sure what it will render when the post gets merged, though.

Reviewer: @huachaohuang to update the create date, it's easy to delete and recreate with identical content I think :P

Reviewer: > what it will render when the post gets merged

I guess not. Git will keep file's stat.

Author: Hmm, maybe we just keep it until we have a better solution. It's easier to reason about if we control it explicitly.

Reviewer: Yes. That makes sense.

> Today, we're excited to announce the release of Engula 0.2! Since [demo 1](posts/demo-1), Engula has entered a formal development cycle, and Engula 0.2 is the first official release. Engula is a serverless storage engine that aims to help users build reliable and cost-effective databases.

Reviewer: ```suggestion
Today, we're excited to announce the release of Engula 0.2! Since [demo 1](demo-1.md), Engula has entered a formal development cycle, and Engula 0.2 is the first official release. Engula is a serverless storage engine that aims to help users build reliable and cost-effective databases.
```

Please verify links locally.

## What's in 0.2?

**We have published a [design document][design-document] to introduce Engula's concepts and architecture.** For those who don't know much about Engula or haven't been around for a while, Engula has changed a lot, and the design document is the best place to get started.

[design-document]: https://github.com/engula/engula/blob/main/docs/design.md

Engula 0.2 comes with one engine and three kernels:

- **A hash engine that supports simple key-value operations.** This engine relies on a kernel to perform stateful operations. It can work with the three kernels below to tackle different use cases.
- **A memory kernel that stores everything in memory.** This kernel integrates a memory journal, storage, and manifest for data storage.
- **A file kernel that stores everything in local files.** This kernel integrates a file journal, storage, and manifest for data storage.
- **A gRPC kernel that stores data in remote gRPC servers.** An engine uses a kernel client to communicate with a kernel server. The kernel server can further connect to a journal server and a storage server for data storage. Engula 0.2 provides a binary to start different kinds of servers easily.

> **If you want to try it out, we have prepared a [tutorial](/posts/tutorial-0.2.md) for you. Have fun!**

Reviewer: ```suggestion
**If you want to try it out, we prepare a [tutorial](tutorial-0.2.md) for you. Have fun!**
```

Reviewer: The link is broken.

> 
> ## What's next in 0.3?
> 
> TODO

Reviewer: ```suggestion
```

per https://github.com/engula/engula.github.io/pull/17#issuecomment-994925017

## Community development

> Engula is an ambitious project that tries to offer a solid database infrastructure for the next decade. It requires a lot of effort and different kinds of skills. We believe that a supportive community is vital to the success of Engula, and we are glad to see that the community is going in a promising direction.

Reviewer: "a solid database infrastructure" is ambiguous of a database, while we state engula is a storage engine. We'd better keep consistent reference so that users get a repeatedly reinforced impression.

If you are interested in joining our community, we have some guidelines for you:

> - Please read [the code of conduct][coc] first.

Reviewer: The CONTRIBUTING guide can be more proper here.
